invenio_records_resources/records/systemfields/files/manager.py

```diff
@@ -266,24 +270,61 @@ def teardown(self, full=True):
         self._order = []
 
     def copy(self, src_files, copy_obj=True):
-        """Copy from another file manager."""
+        """Copy from another file manager.
+
+        This method will copy all object versions to the `self.bucket` assuming
+        that the latter is a new empty bucket.
+        """
         self.enabled = src_files.enabled
 
         if not self.enabled:
             return
 
-        for key, rf in src_files.items():
-            # Copy object version of link existing?
-            if copy_obj:
-                dst_obj = rf.object_version.copy(bucket=self.bucket)
-            else:
-                dst_obj = rf.object_version
-
-            # Copy file record
-            if rf.metadata is not None:
-                self[key] = dst_obj, rf.metadata
-            else:
-                self[key] = dst_obj
+        bucket_objects = ObjectVersion.query.filter_by(bucket_id=self.bucket_id).count()
+        if bucket_objects < 1:
```

Comment:
```suggestion
        if bucket_objects == 0:
```
i would be worried if we have -1